packages/store/src/StoreCore.sol

```diff
@@ -302,6 +305,20 @@ library StoreCore {
   ) internal {
     uint256 location = StoreCoreInternal._getStaticDataLocation(tableId, keyTuple);
 
+    // Emit event to notify offchain indexers
+    emit StoreCore.StoreSpliceStaticData({
+      tableId: tableId,
+      keyTuple: keyTuple,
+      start: start,
+      deleteCount: deleteCount,
+      data: data
+    });
+
+    // Early return if the table is an offchain table
+    if (tableId.getType() == RESOURCE_OFFCHAIN_TABLE) {
+      return;
+    }
```

Comment:
do we support splicing for offchain tables?

Reply:
we do for static fields! Since splicing for static fields doesn't require reading anything from storage, can just support it for static fields without any further changes. For dynamic fields it's not supported because it would require reading the dynamic field's length from storage (but it's not stored in storage), see https://github.com/latticexyz/mud/pull/1558#discussion_r1332249965

Codegen doesn't render the dynamic field methods for offchain tables, but does render all other ones, so much more "real" tables than before with ephemeral

Reply:
ohh I see, I was thinking about push/pop but that's a dynamic field thing 👌